BrowserKit/Tests/WebEngineTests/WKEngineSessionTests.swift

```diff
@@ -403,6 +403,32 @@ final class WKEngineSessionTests: XCTestCase {
         XCTAssertEqual(contentScriptManager.uninstallCalled, 1)
     }
 
+    // MARK: WKEngineWebViewDelegate
+
+    func testFindInPageGivenSelectionThenCallsDelegate() {
+        let subject = createSubject()
+        subject?.delegate = engineSessionDelegate
+        let expectedSelection = "A search"
+
+        subject?.tabWebView(webViewProvider.webView, findInPageSelection: expectedSelection)
+
+        XCTAssertEqual(engineSessionDelegate.findInPageCalled, 1)
+        XCTAssertEqual(engineSessionDelegate.savedFindInPageSelection, expectedSelection)
+        prepareForTearDown(subject!)
+    }
```

Comment:
Do these tests actually exercise any code running in the client? It feels like in practice some of our mocks right now only test their own artificial code that runs only on CI. (This is a broader topic not specific just to this PR but I'm wondering if it's something that would be worth us revisiting at some point.)

Reply:
In the current case, the real object `WKEngineSession` get tested, but everything around it are Mocks so we can control the output of the test. `WKEngineSession` is not used in any parts under the client app as of now thought. Not sure if this answers the question you asked?

Reply:
Ok thanks @lmarceau for clarifying 👍 I was just wondering how much value we get out of some of our mocks running on CI (again, not specific to this PR, more of a general question). LGTM